lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -246,6 +260,20 @@ type instance AllowedMnemonics 'Ledger    = '[12,15,18,21,24]
 type instance AllowedMnemonics 'Shelley   = '[15,18,21,24]
 type instance AllowedMnemonics 'SndFactor = '[9,12]
 
+
+-- | Machinery to demote type level [Nat] to a normal [Integer] list
+--
+-- Needed to get the result of @AllowedMnemonics@ as a value.
+class ManyNatVal (xs :: [Nat]) where
+  manyNatVal :: proxy xs -> [Integer]
+
+instance ManyNatVal '[] where
+  manyNatVal _ = []
+
+instance (KnownNat a, ManyNatVal as) => ManyNatVal (a ': as) where
+  manyNatVal _ =
+    natVal (Proxy :: Proxy a) : manyNatVal (Proxy :: Proxy as)
```

Comment:
I just remembered that this already exists as `NatVals` in module Cardano.Wallet.Primitive.AddressDerivation